FSCalendar/FSCalendar.m

```diff
@@ -1278,6 +1284,18 @@ - (void)invalidateDateTools
     _formatter.calendar = _gregorian;
     _formatter.timeZone = _timeZone;
     _formatter.locale = _locale;
+
+    [self updateToday];
+}
+
+- (void)updateToday
+{
+    NSDateComponents *dateComponents = [self.gregorian components:(NSCalendarUnitYear|NSCalendarUnitMonth|NSCalendarUnitDay) fromDate:[NSDate date]];
+    dateComponents.hour = 0;
+    dateComponents.minute = 0;
+    dateComponents.second = 0;
+    dateComponents.timeZone = self.timeZone;
```

Comment:
So this is the key part. and we reset it when today is set?

and is `invalidateDateTools` the local method to reset the date/time. when does that fire?

Reply:
it only fires during the setup, when we do something like `calendarView.timeZone = vehicle.timeZone`

Reply:
I took the code from this open PR in their repo, https://github.com/WenchaoD/FSCalendar/pull/998/files and tested that the calendar still works on search and vehicle page